hkmc2/shared/src/main/scala/hkmc2/semantics/ucs/DesugaringBase.scala

```diff
@@ -77,7 +77,7 @@ trait DesugaringBase(using Ctx, State):
     app(matchResultClass, tup(fld(captures)), "result of `MatchResult`")
     
   protected final def makeMatchFailure =
-    app(matchFailureClass, tup(), "result of `MatchFailure`")
+    app(matchFailureClass, tup(fld(Term.Lit(UnitLit(true)))), "result of `MatchFailure`")
```

Comment:
wow i didn't know it could be fixed by simply adding a default argument

thanks Luyu